src/hooks/useModalControl.ts

```diff
@@ -0,0 +1,46 @@
+import routes from '@constants/routes';
+import useDeleteGoal from '@hooks/api/goalsAPI/useDeleteGoal';
+import { useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+const useModalControl = () => {
+  const [showTodoModal, setShowTodoModal] = useState(false);
+  const [isDeletePopupVisible, setIsDeletePopupVisible] = useState(false);
+  const [goalId, setGoalId] = useState<number>(0);
+  const { mutate: deleteGoalMutate } = useDeleteGoal();
+  const handleShowTodoModal = () => setShowTodoModal(true);
+  const handleShowDeletePopup = (id: number) => {
+    setIsDeletePopupVisible(true);
+    setGoalId(id);
+  };
+  const handleCloseTodoModal = () => setShowTodoModal(false);
+  const handleCloseDeletePopup = () => setIsDeletePopupVisible(false);
+  const navigate = useNavigate();
+  const location = useLocation();
+  const handleDelete = () => {
+    deleteGoalMutate(goalId, {
+      onSuccess: () => {
+        setIsDeletePopupVisible(false);
+        if (
+          location.pathname === `/${routes.goalDetail}/${goalId}` ||
+          location.pathname === `/notes/${goalId}`
+        ) {
+          navigate(`/${routes.dashboard}`);
```

Comment:
```suggestion
        if (
          location.pathname === `${routes.goalDetail}/${goalId}` ||
          location.pathname === `${routes.notes}/${goalId}` ||
          location.pathname === `${routes.newNote}/${goalId}`
        ) {
          navigate(`${routes.dashboard}`);
```
앞에 / 빼고 newnote쪽도 추가 해주실 수 있을까용???